### ISSUE TITLE
Enable Netlify CMS "Open Authoring" feature

### ISSUE THREAD

**Issue by JohnVeness** [xbmc/kodi-tv#304]
TL;DR: Can you investigate the Netlify CMS "Open Authoring" feature (https://www.netlifycms.org/docs/open-authoring/) to enable ordinary users to *suggest* edits via the CMS?

Long version:

As mentioned in #300, I noticed several blog posts which could be improved, in formatting, or language/spelling/grammar. It was mentioned, quite rightly, that it is preferred for changes to be made via the CMS rather than raw editing the files. I decided to have a play for my own interest, and managed to clone the site into my own Netlify CMS instance, and test editing files, which went quite well.

Rather than me asking if I could have CMS access to make these little changes (which would be unlikely as I'm basically a nobody here!), I see that Netlify CMS has a feature they call "Open Authoring", described at https://www.netlifycms.org/docs/open-authoring/ which seems like a great solution. It would allow people like me to make *suggested* changes to pages, done via the CMS (so hopefully won't break anything), which the Kodi team could decide whether or not to accept. And without you needing to trust people with CMS access. Can you look into this?

Footnote: If I *was* to try and sell myself to you, I would point you at three pages I changed an examples of the sort of edits I'd like to make, namely:

- Kodi 15.1 Isengard - Release Candidate: [original](https://kodi.tv/article/kodi-151-isengard-release-candidate/), versus [with my suggested edits](https://main--nimble-salamander-fb6e44.netlify.app/article/kodi-151-isengard-release-candidate/)
- Kodi 15.0 skin showcase: [original](https://kodi.tv/article/kodi-150-skin-showcase/), versus [with my suggested edits](https://main--nimble-salamander-fb6e44.netlify.app/article/kodi-150-skin-showcase/)
- Kore - The new official Android remote: [original](https://kodi.tv/article/kore-new-official-android-remote/), versus [with my suggested edits](https://main--nimble-salamander-fb6e44.netlify.app/article/kore-new-official-android-remote/)

I'm not a professional writer or copy editor or developer, just an end-user (since original Xbox days) of XBMC/Kodi, and someone who likes correcting typos on Wikipedia etc. :)

Thanks for reading.

**Comment by ProfYaffle** [comment]
I'm unsure of the best mechanism to pick up issues like these - I think, so far, they're all related to imperfect import/conversion when we switched from Wordpress to Drupal to Netlify CMS. I'm happy to pick up any specifics and fix them, as I'm usually the face behind the blog posts... whether that's sufficient, or just a duplication of effort to both describe and then fix the problem, though...

I'm certainly happy to accept corrections in some capacity. I'll have a look at those specific posts now.

**Comment by JohnVeness** [comment]
I've been in the position before where I've tried to explain problems with blog posts (on another site), where I had to say "In paragraph three, second sentence, third word should be 'lemon'" or whatever, and it's just frustrating for all involved!

That's why I liked the sounds of the Open Authoring thing, as I could show exactly what I'm suggesting, and you (or someone) can just one click accept/reject it.

**Comment by ProfYaffle** [comment]
Yeah, I'm not going to disagree - it's far easier to say "here's the new text/md" than it is to describe a roadmap to the changes - "out of the door, line on the left, one bullet point each" territory. Having just gone through those three samples for you, I'd probably have to conclude:

1. I inevitably missed some changes - when I published, I could spot the odd missing apostrophe or similar that you'd caught and I hadn't noticed in the comparison.
2. I'm redoing the work you've already done (e.g. the image URLs), which is clearly a Very Silly Thing to Do.
3. I'm not sure how much time it's worth spending correcting historical translation errors (either the human linguistic error type or the code conversion type) on posts for versions we neither support nor provide any more.

We'd really need a markup or redline function, so the changes are obvious; failing that, let's see if there's a better way, such as the one you suggest (I have no knowledge here, so can't prejudge). The pedant in me wants every historical post to be perfect; the realist in me doesn't really want to volunteer for that, or thinks that it's necessarily, well, necessary.

Many thanks for the contributions. If you see stuff that's glaringly wrong in more current posts, shout and I'll update as required.

**Comment by JohnVeness** [comment]
> The pedant in me wants every historical post to be perfect; the realist in me doesn't really want to volunteer for that, or thinks that it's necessarily, well, necessary.

I agree that it's probably not very important, but, well, I'm volunteering to fix historical things (if we can make it easy for each other)!

As far as I can tell, enabling Open Authoring is just one line or two to add to the site config.

**Comment by pkscout** [comment]
One or two lines in the site config, and and entirely new workflow someone will have to manage.

According to the docs, Open Authoring forks the entire website repo, then the outside author makes the changes in that fork, and then it generates a PR that has to be reviewed **in Github** and then merged.  Which means it's really no different than if you manually changed the markdown files and submitted a PR.  It gives the outside author the ability to use the Netlify CMS as an editor, but no ability for a team member to check the work using the CMS.  It's all manual review of markdown files and looking at everything in the preview deploy (which someone will have to manually approve in Netlify since we don't allow deploy previews to be built automatically from outside PRs).

That will have to be done against staging.  We will not accept PRs against main.

**Comment by JohnVeness** [comment]
Thanks. Yes, I realise I was probably oversimplifying the entire process, even if the setup is "easy".

Regarding previewing, I could provide a link to how an edited page looks on my cloned Netlify instance in the PR, if such a thing is not provided automatically. I'm happy to work in staging or wherever you wish.

In any case, I'm trying to help, not to be a burden, so of course this is up to you :)

**Comment by ProfYaffle** [comment]
I don't have enough knowledge to comment knowledgeably about the workflow implications of "open authoring" or similar, although I'm inclined to side with @pkscout simply because he does.

As an alternative, I wonder if there's some clear markdown - maybe a diff or just an extract of a diff - that we could use. If I can clearly see the changes, I have no problem merging them in, it's just that I don't want to hunt-and-peck alterations in multiple antediluvian posts. I would genuinely be interested where posts are materially dysfunctional, such as the essay spam or broken image links you picked up - I'm less bothered about translation errors or style in historical posts unless a post needs editing anyway.
